plone/app/textfield/transform.py

```diff
@@ -67,3 +72,26 @@ def __call__(self, value, mimeType):
             # log the traceback of the original exception
             LOG.error("Transform exception", exc_info=True)
             raise TransformError('Error during transformation', e)
+
+    def check_referenced_images(self, target_mimetype, cache_obj):
+        # extract all image src uuids
+        uids = re.findall('src="[^/]*/resolve[uU]id/([^/"]*)', cache_obj.value)
+        if len(uids) == 0:
+            # no uuid here at all
+            return
+
+        # referenced image scale urls get outdated if the images are modified.
+        # purging the transform cache updates those urls.
+        cache = Cache(cache_obj, context=self.context)
+        data = cache.getCache(target_mimetype)
+        if data is None:
+            # data is not cached
+            return
+
+        # get the original save time from the cached data dict
+        orig_time = getattr(cache_obj, cache._id).values()[0][0]
```

Comment:
I would really like to see a catalog counter check (``portal_catalog.getCounter()``) here, which implies that we write the prior counter to some place, like a volatile attribute on the content object to be checked. If the count stays the same there is IMO no need for further checks.

Reply:
I've implemented the catalog counter check and saved it to the `cache_obj` like PortalTransform.cache does

Reply:
ok, I missed that one.